logo.js

```diff
@@ -76,6 +76,26 @@ function LogoInterpreter(turtle, stream, savehook)
     return atom === match;
   }
 
+  function isPromise(value) {
+    return value && value.then && value.catch;
+  }
+
+  function promiseFinally(promise, finalBlock) {
```

Comment:
I think this can be simplified to:

```js
function promiseFinally(promise, finalBlock) {
  return promise.then(finalBlock, finalBlock).then(function() { return promise; });
}
```